R/as-json.R

```diff
@@ -36,3 +36,7 @@ method(as_json, list(Provider, TypeArray)) <- function(provider, x) {
     items = as_json(provider, x@items)
   )
 }
+
+method(as_json, list(Provider, TypeJsonSchema)) <- function(provider, x) {
```

Comment:
We'll need to do a bit more if we also want this to work with `tool`/`ToolDef` (which I think we do for the sake of consistency). shis will need a little extra work in order if we also want to allow json schema in tool specs. That will require some work on the UI side at `tool()` (since currently you supply named arguments) and for the `as_json(*, ToolDef)` methods.